src/credential/credential-internal.ts

```diff
@@ -0,0 +1,472 @@
+/*!
+ * Copyright 2020 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Use untyped import syntax for Node built-ins
+import fs = require('fs');
+import os = require('os');
+import path = require('path');
+
+import { GoogleOAuthAccessToken, Credential } from './credential';
```

Comment:
These 2 types should probably in a separate file to avoid the cyclic dependency between `credential` and `credential-internal`. Or move the credential factory functions to a separate file.